sklearn/cluster/dbscan_.py

```diff
@@ -301,8 +301,8 @@ class DBSCAN(BaseEstimator, ClusterMixin):
     and Data Mining, Portland, OR, AAAI Press, pp. 226-231. 1996
     """
 
-    def __init__(self, eps=0.5, min_samples=5, metric='euclidean',
-                 metric_params=None, algorithm='auto', leaf_size=30, p=None,
+    def __init__(self, eps=0.5, min_samples=5, metric='minkowski',
+                 metric_params=None, algorithm='auto', leaf_size=30, p=2,
```

Comment:
A warning needs to be issued if the user has not set metric but has set p. You will need to use dummy values here during a temporary deprecation period. See our contributor guide on changing default values

Reply:
Hello @jnothman 
Thank you!
For example I'll use this:


		if metric == 'minkowski' and p != 2:
			warnings.warn('something')
			p = 2

That's it?

Reply:
@gilbertoolimpio it's more that you need to set the default value to "warn" for instance, and then in the `fit` function for instance, if it's "warn", warn the user, and set the actual default value. See [here](http://scikit-learn.org/dev/developers/contributing.html#change-the-default-value-of-a-parameter)

Reply:
@adrinjalali 
Tks!